tests/suites/os/tests/internet-sharing/index.js

```diff
@@ -0,0 +1,110 @@
+                //
+                // A racing condition between balenaEngine and NetworkManager led to
+                // some Internet connection sharing iptables rules not being applied
+                // when NM connection profile with Internet sharing was activated at boot.
+                //
+                // This test checks whether all necessary iptables rules are added by
+                // artificially blocking iptables for 1 second and then for 3 seconds
+                // while a connection profile with Internet sharing is activated.
+
+                // Counts all iptables rules set for Internet sharing by NM
+                async function countSharedIptables(that) {
+                    // Executes a command that returns an integer
+                    async function execInt(command, that) {
+                        const output = await that.worker.executeCommandInHostOS(
+                            command,
+                            that.link
+                        );
+                        return parseInt(output);
+                    }
+
+                    let count = await execInt(
+                        'iptables -S | grep -c nm-shared-dummy0 || true',
+                        that,
+                    );
+                    count += await execInt(
+                        'iptables -S -t nat | grep -c nm-shared-dummy0 || true',
+                        that,
+                    );
+                    count += await execInt(
+                        'iptables -S nm-sh-fw-dummy0 | wc -l',
+                        that,
+                    );
+                    count += await execInt(
+                        'iptables -S nm-sh-in-dummy0 | wc -l',
+                        that,
+                    );
+
+                    that.log(`iptables count ${count}`);
+
+                    return count;
+                }
+
+                // Create a dummy NM connection with Internet sharing enabled. 
+                await this.worker.executeCommandInHostOS(
+                    'nmcli c add type dummy ifname dummy0 con-name dummy' + 
+                    ' autoconnect no ipv4.method shared ipv6.method disabled',
+                    this.link
+                );
+
+                // Lock iptables for 1 second and activate the connection with sharing
+                // enabled in parallel. NM waits for 2 seconds for the lock to be released
+                // and at the end of the one second artificial lock it will be able to 
+                // set successfully all iptables rules. Previously that would fail.
+                await this.worker.executeCommandInHostOS(
+                    'flock /run/xtables.lock sleep 1 & nmcli c up dummy & wait',
+                    this.link
+                );
+                let count = await countSharedIptables(this);
+                test.is(count, 14, 'Internet sharing iptables rules are all set');
```

Comment:
Instead of using `test.is`, you can try using a hard equal assertion called `test.equal`. 
Docs: https://node-tap.org/docs/api/asserts/#tequalfound-wanted-message-extra

Reply:
Changing this now as well.